### PR TITLE
kafka: FIXES SendImageNotification.

### PR DESCRIPTION
**Pull request by ldjebran**
# Description
* Use the global producer for SendImageNotification as it has the right configuration.
* Fixes SendDeviceNotification as producer maybe nil.
* added unitests for both fixes.

FIXES: THEEDGE-3000

## Type of change

What is it?

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [X] Tests update
- [ ] Refactor

# Checklist:

- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I run `make pre-commit` to check fmt/vet/lint/test-no-fdo

### CODE REVIEW

File: pkg/services/errors.go [hunk 1]
```diff
@@ -374,6 +375,13 @@ func (e *KafkaAllBrokersDown) Error() string {
 	return KafkaAllBrokersDownMsg
 }
 
+// KafkaProducerInstanceUndefined indicates that we were not able to get a kafka producer instance
+type KafkaProducerInstanceUndefined struct{}
+
+func (e *KafkaProducerInstanceUndefined) Error() string {
```
Comment (omaciel) [inline pkg/services/errors.go:381]: Quick line to `pkg/services/errors_test.go` will make you friends :)

File: pkg/services/errors.go [hunk 2]
```diff
@@ -374,6 +375,13 @@ func (e *KafkaAllBrokersDown) Error() string {
 	return KafkaAllBrokersDownMsg
 }
 
+// KafkaProducerInstanceUndefined indicates that we were not able to get a kafka producer instance
```
Comment (mgold1234) [inline pkg/services/errors.go:378]: we are not able to get?

Comment (ldjebran) [reply pkg/services/errors.go:378]: what I can confirm is that we were not (or at that moment in the past we were not ) able to get a producer,  I cannot affirm that generally and now I cannot get a producer.

File: pkg/services/errors.go (hunk 1 above)
Comment (ldjebran) [reply pkg/services/errors.go:381]: Thanks for the hint, added a line.

File: pkg/services/images.go [hunk 3]
```diff
@@ -1611,17 +1606,14 @@ func (s *ImageService) SendImageNotification(i *models.Image) (ImageNotification
...
 			Value:          []byte(recordValue),
-		}, nil)
-
-		if perr != nil {
-			s.log.WithField("message", perr.Error()).Error("Error on produce")
+		}, nil); err != nil {
+			s.log.WithField("message", err.Error()).Error("Error on produce")
 			return notify, err
 		}
-		p.Close()
```
Comment (omaciel) [inline pkg/services/images.go:1624]: You don't need to close it anymore?

**Review by omaciel — APPROVED** [review]
/lgtm

File: pkg/services/images.go (hunk 3 above)
Comment (ldjebran) [reply pkg/services/images.go:1624]: we do need to close as we are using a global producer.

**Review by omaciel — APPROVED** [review]
/lgtm

**Comment by adeliaferreira** [comment]
LGTM

**Review by de1987 — APPROVED** [review]

**Review by mgold1234 — APPROVED** [review]
/lgtm

**Pull request merged**